### PR TITLE
chore(api): new output api #6677

### PR DESCRIPTION
**Pull request by amcdnl**
Given all the back and forth on #7161, I decided to open a new clean PR with the changes. This PR does the following:

- Adds `opened` event, deprecating `open`
- Adds `closed` event, deprecating `close`
- Adds `openChange<boolean>`

### CODE REVIEW

File: src/lib/sidenav/drawer.ts [hunk 1]
```diff
@@ -192,11 +194,38 @@ export class MatDrawer implements AfterContentInit, OnDestroy {
...
+  /** Event emitted when the drawer open state is changed. */
+  @Output() openedChange: EventEmitter<boolean> = new EventEmitter<boolean>();
 
-  /** Event emitted when the drawer is fully closed. */
-  @Output('close') onClose = new EventEmitter<MatDrawerToggleResult | void>();
+  /** Event emitted when the drawer has been opened. */
+  @Output('opened')
+  get _openedStream(): Observable<void> {
+    return RxChain.from(this.openedChange)
+      .call(filter, o => o)
```
Comment (mmalerba) [inline src/lib/sidenav/drawer.ts]: nit: indent continuation lines 4 spaces

File: src/lib/sidenav/drawer.spec.ts [hunk 2]
```diff
@@ -441,8 +441,8 @@ class DrawerContainerTwoDrawerTestApp {
   template: `
     <mat-drawer-container (backdropClick)="backdropClicked()">
       <mat-drawer #drawer position="start"
-                 (open)="open()"
-                 (close)="close()">
+                 (opened)="open()"
+                 (closed)="close()">
```
Comment (mmalerba) [inline src/lib/sidenav/drawer.spec.ts]: can we add a test for the `[(opened)]` syntax as well

**Comment by kara** [comment]
@amcdnl Rebase?

**Comment by amcdnl** [comment]
@kara - done :)

**Comment by kara** [comment]
One more time?   :-)

**Comment by amcdnl** [comment]
@kara - done :L

**Pull request merged**